include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -824,6 +836,20 @@ class __future : private std::tuple<_Args...>
         auto new_tuple = std::tuple_cat(new_val, (std::tuple<_Args...>)*this);
         return __future<_Event, _T, _Args...>(__my_event, new_tuple);
     }
+
+    template <typename _T, std::size_t _NResults>
+    static __future<_Event, std::shared_ptr<__result_and_scratch_storage_base>>
+    __convert_to_future_with_scratch_storage_base_ptr(
```

Comment:
Why do we need `__convert_to_future_with_scratch_storage_base_ptr`?
It was not before.
In any case `__convert_to_future_with_scratch_storage_base_ptr` looks complicated and overall the change is not for 2022.9.0 milestone I think.

Reply:
I think it may be discussion question.
Let's ask @timmiesmith, @danhoeflinger and @mmichel11 about this.